docs/rfcs/UserProfilesv2.md

```diff
@@ -0,0 +1,165 @@
+- Must provide structures that allow to generate backward-compatible profiles (v1 profiles)
+- Add structured access information that can be reliably queried
+- Integrate the [Automatic Access Expiration](AutomaticAccessExpiration.md) to the access information
+- Allow for identity information to be transported from social authentication providers to our profile
+- Standardize documentation around the profile
+- Allow for a standardized API on top of the user profile data ([Person-API](https://github.com/mozilla-iam/person-api))
+- Version the profile for possible future updates
+- Field-level metadata support
+  - Support cryptographic signature per field
+  - Support data classification per field
+  - Support publisher identifier per field
+
+## Use cases
+
+1. Machines such as TaskCluster want to reliably query Person-API and get authoritative answers about the user's group
+   membership. The groups must be easy to query with no ambiguity (i.e. no string-namespacing).
+2. User managing GitHub wants to access GitHub user_ids (not email) from logged in users in order to query the GitHub
+   API and get relevant data about the user.
+3. The access provider wants to expire access when it is no longer used (Automatic Access Expiration).
+4. Security team wants to assert that access is lost to certain services (RPs) when they're no longer used by the person
+   who used to have access.
+5. CIS wants to validate if a publisher has authority over the fields it modified.
+6. An RP wants to get profile attributes that are considered public-only, while another, more privileged RP wants to get
+   profile attributes that are considered STAFF CONFIDENTIAL.
+7. An RP wants to strongly verify that the profile data it's consuming has not been modified anywhere in the IAM
+   pipeline and represent what the publisher initially asserted.
+
+
+## Possible corresponding API
+
+### Classification scope
+
+**Scopes**: `classification:public`, `classification:staff`, `classification:workgroup`, `classification:individual`
+
+Only returns fields which `metadata.classification` matches the corresponding scope, regardless of ANY other scopes.
+Defaults to `classification:public` regardless of the scope selection (i.e. including `classification:public` is
+optional for the API caller)
+
+### Publisher Authority scope
+
+**Scopes**: `authority:mozilliansorg`, `authority:auth0`, `authority:hris`, `authority:ldap`
+
+Only authorized for specific publishers. Generally only one type of publisher is allowed one of these scopes (i.e. LDAP
+cannot be authorized both `authority:auth0` and `authority:ldap` for example). However, a single publisher may have
+multiple, separate pieces of code accessing the API with different tokens and scopes, yet the same `authority` specific
+scope (i.e. Mozillians.org may have 3 `client_id` accessing the authorizer with the same `authority:mozilliansorg`
+scope).
+
+### Endpoints
+
+- GET,PATCH,PUT,DELETE /v2/{user_id}
```

Comment:
true
i think we should add a different endpoint for batch.
i think there's already some paging implementation in person-api which we would leverage, it might not be documented (i based this on the current person-api readme)

Possible example:

- GET, PATCH,PUT `/v2/users?q=*&offset=0&perpage=100`
Paginated, searchable query
Scopes: same as `/v2/{user_id}`
Parameters: 
- `q` query string (syntax?)
- `offset` page position to start from
- `perpage` number of pages (eg `offset=0&perpage=100` returns the first 100 users. `offset=100&perpage=10` returns user 101 to 111. Can also use `position` if `offset` is confusing, same usage but starts at `)

Body:
validated user profile, same requirements as `/v2/{user_id}`